.github/workflows/frontend-verify.yml

```diff
@@ -0,0 +1,122 @@
+name: "Frontend Lint & Compile check"
+
+on:
+  push:
+    paths:
+      - "frontend/**"
+  pull_request:
+    paths:
+      - "frontend/**"
+
+jobs:
+  lint-and-compile:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Cache node modules
+        id: cache-npm
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('frontend/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18.10.0'
+      - name: Install dependencies 
+        working-directory: frontend
+        run: npm ci
+      - name: Run lint
+        working-directory: frontend
+        run: npm run lint
+      - name: Run build frontend
+        working-directory: frontend
+        run: |
+          REACT_APP_BUSINESS=true 
```

Comment:
Not sure if we want to run thefrontend in enterprise mode. I think we are running against the community backend version for the tests

Reply:
Removed this. @rikimaru0345 to confirm it's the right action.

Reply:
Yes, normal, non-business mode is fine.
It only introduces extra steps that we don't need.